sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2106,6 +2106,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val OPTIMIZER_IGNORE_HINTS =
+    buildConf("spark.sql.optimizer.ignoreHints")
```

Comment:
I'm the one who is still confused about the conf name.
@cloud-fan , is it okay not to use `.enabled` for this new boolean configuration?

Reply:
@dongjoon-hyun `spark.sql.ignoreHints.enabled` kind of sounds a little odd isn't it ? But if thats the consensus then i will change it.

Reply:
For that, I'm not sure either. That's the reason why I pinged @cloud-fan to get his advice.

BTW, I'm okay for both `spark.sql.optimizer.xxx` or `spark.sql.xxx`. As I replied in a different comment, I asked to change only the variable name `val OPTIMIZER_IGNORE_HINTS`.